src/main/java/net/masterthought/cucumber/reducers/ReportFeatureWithRetestMerger.java

```diff
@@ -0,0 +1,103 @@
+package net.masterthought.cucumber.reducers;
+
+import net.masterthought.cucumber.json.Element;
+import net.masterthought.cucumber.json.Feature;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Merge list of given features. If there are couple of scenarios with the same Id then
+ * only the latest will be stored into the report.
+ *
+ * Uses when need to generate a report with rerun results of failed tests.
+ */
+final class ReportFeatureWithRetestMerger implements ReportFeatureMerger {
+
+    private static final String ERROR = "You are not able to use this type of results merge. The start_timestamp field" +
+            " should be part of element object. Please, update the cucumber-jvm version.";
+    private static final ElementComparator ELEMENT_COMPARATOR = new ElementComparator();
+
+    @Override
+    public List<Feature> merge(List<Feature> features) {
```

Comment:
there are some gaps here
https://codecov.io/gh/damianszczepanik/cucumber-reporting/pull/894/diff#D2-40

Reply:
@dabogee cover this